Build failure: `obs-studio`

### Steps To Reproduce

Steps to reproduce the behavior:
1. build *obs-studio*

### Build log

```
❯ sudo nixos-rebuild switch --upgrade-all
unpacking channels...
unpacking channels...
unpacking channels...
building Nix...
building the system configuration...
these 14 derivations will be built:
  /nix/store/y2342xfbbvvgynb19ynqqj1fq3ydsa34-obs-studio-29.0.2.drv
  /nix/store/mr075kkpanl3568y336ydvbc3a7d6zsy-system-path.drv
  /nix/store/vcqbldk9d1xr87605b83i4jgqp21vach-dbus-1.drv
  /nix/store/0kp9iy066l58clj5sp4p6dv7fcvqwz9f-X-Restart-Triggers.drv
  /nix/store/27rdh4527likg28jj2hpikdv11v4gg2n-unit-systemd-fsck-.service.drv
  /nix/store/9580wkhcixfj4d1v8728bfcd67j1qcpa-X-Restart-Triggers.drv
  /nix/store/c4gnci9a2p3ps949wav6lyalfiag65lw-unit-dbus.service.drv
  /nix/store/ilxb9xzp0j0kr8y9sxckc6ac726sml2m-unit-polkit.service.drv
  /nix/store/nsm8w7dakad744fdh10b1252gqnyglid-unit-accounts-daemon.service.drv
  /nix/store/caswr54ma8kd6qhi0q4g2rbhzz55cvvy-system-units.drv
  /nix/store/qgqwzdkzksy5rfl0ywsvvpw755i2qlq8-unit-dbus.service.drv
  /nix/store/x41d4zfx5df0impywsisf0j5fcc5sj4m-user-units.drv
  /nix/store/p1s7bzm7nm4ccihh0dsgf3bx3f9l8k1i-etc.drv
  /nix/store/qi0spjkhk8g7yizjirp3mw0hafq2z39j-nixos-system-Simon-s-Tuxedo-InfinityBook-14-Gen8-23.05.5125.0561103cedb1.drv
building '/nix/store/y2342xfbbvvgynb19ynqqj1fq3ydsa34-obs-studio-29.0.2.drv'...
qtPreHook
unpacking sources
unpacking source archive /nix/store/n0b6gvg9dy007gzb3v5ykymwzzp0fvbl-source
source root is source
patching sources
applying patch /nix/store/f4kv9vm7iyvq8k725hhypqpsdnkzxhzn-Enable-file-access-and-universal-access-for-file-URL.patch
patching file plugins/obs-browser/obs-browser-source.cpp
Hunk #1 succeeded at 224 with fuzz 2 (offset 45 lines).
applying patch /nix/store/4nazi4kng4qp69r9zxfaw02ppxrczg72-Provide-runtime-plugin-destination-as-relative-path.patch
patching file libobs/CMakeLists.txt
Hunk #1 succeeded at 458 (offset 3 lines).
patching file libobs/obsconfig.h.in
Hunk #1 succeeded at 13 with fuzz 1.
configuring
fixing cmake files...
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/jlqkllv8ndixip9g0j1h991lxi7y32h1-obs-studio-29.0.2/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/jlqkllv8ndixip9g0j1h991lxi7y32h1-obs-studio-29.0.2/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/jlqkllv8ndixip9g0j1h991lxi7y32h1-obs-studio-29.0.2/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/jlqkllv8ndixip9g0j1h991lxi7y32h1-obs-studio-29.0.2/share/doc/obs-studio -DCMAKE_INSTALL_INFODIR=/nix/store/jlqkllv8ndixip9g0j1h991lxi7y32h1-obs-studio-29.0.2/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/jlqkllv8ndixip9g0j1h991lxi7y32h1-obs-studio-29.0.2/share/man -DCMAKE_INSTALL_OLDINCLUDEDIR=/nix/store/jlqkllv8ndixip9g0j1h991lxi7y32h1-obs-studio-29.0.2/include -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/jlqkllv8ndixip9g0j1h991lxi7y32h1-obs-studio-29.0.2/include -DCMAKE_INSTALL_SBINDIR=/nix/store/jlqkllv8ndixip9g0j1h991lxi7y32h1-obs-studio-29.0.2/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/jlqkllv8ndixip9g0j1h991lxi7y32h1-obs-studio-29.0.2/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/jlqkllv8ndixip9g0j1h991lxi7y32h1-obs-studio-29.0.2/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_OSX_SYSROOT= -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/3hbxw05vfs6m1155xa6pvribs3bv777n-gcc-wrapper-12.2.0/bin/strip -DCMAKE_RANLIB=/nix/store/3hbxw05vfs6m1155xa6pvribs3bv777n-gcc-wrapper-12.2.0/bin/ranlib -DCMAKE_AR=/nix/store/3hbxw05vfs6m1155xa6pvribs3bv777n-gcc-wrapper-12.2.0/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/jlqkllv8ndixip9g0j1h991lxi7y32h1-obs-studio-29.0.2 -DCMAKE_CXX_FLAGS=-DDL_OPENGL=\"$(out)/lib/libobs-opengl.so\" -DOBS_VERSION_OVERRIDE=29.0.2 -Wno-dev -DBUILD_BROWSER=ON -DCEF_ROOT_DIR=../../cef -DENABLE_JACK=ON 
-- OBS:  Application Version: 29.0.2 - Build Number: 1
-- The C compiler identification is GNU 12.2.0
-- The CXX compiler identification is GNU 12.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/3hbxw05vfs6m1155xa6pvribs3bv777n-gcc-wrapper-12.2.0/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/3hbxw05vfs6m1155xa6pvribs3bv777n-gcc-wrapper-12.2.0/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found OpenGL: /nix/store/gr5qpcswiyvg69xriylihlil1amv7fdi-libGL-1.6.0/lib/libOpenGL.so   
-- Found FFmpeg: /nix/store/xjzvncwg01pvwq7nz08lcfsgibnkg565-ffmpeg-full-5.1.3-lib/lib/../lib/libavcodec.so (found version "#define LIBAVCODEC_VERSION_MINOR  37;#define LIBAVCODEC_VERSION_MICRO 100.37.100") found components: avcodec avdevice avutil avformat 
-- Found CURL: /nix/store/8x8b9370k4g4x18kqdxsqx60zdxivpgv-curl-8.1.1/lib/libcurl.so (found version "8.1.1")  
-- Found Luajit: /nix/store/lx61sy4wmy7qnpfhhdqhch9k6kqddldd-luajit-2.1.0-2022-10-04/lib/../lib/libluajit-5.1.so  
-- Found SWIG: /nix/store/p6dsra0n8877w62nsx7rq241v56ic1pk-swig-3.0.12/bin/swig (found suitable version "3.0.12", minimum required is "3")  
-- OBS:  ENABLED    obslua
-- Found Luajit: /nix/store/lx61sy4wmy7qnpfhhdqhch9k6kqddldd-luajit-2.1.0-2022-10-04/lib/libluajit-5.1.so  
-- OBS:  obs-scripting -> Luajit found.
-- Found Python: /nix/store/n5fr2ppq4i3hdjbbf4636gk54s07022v-python3-3.10.13/bin/python3.10 (found version "3.10.13") found components: Interpreter Development Development.Module Development.Embed 
-- OBS:  ENABLED    obspython
-- OBS:  obs-scripting -> Python 3.10.13 found.
-- OBS:  Using system Jansson library.
-- Found X11: /nix/store/9r9i76xm11r0m5q57n01j0zn3ks6lms4-xorgproto-2021.5/include   
-- Looking for XOpenDisplay in /nix/store/p4i93n4m7al2anavcz54cybw9hy2pik3-libX11-1.8.7/lib/libX11.so;/nix/store/drnbcwzvha7mrb5yxzyrdy9a75c4gjgq-libXext-1.3.4/lib/libXext.so
-- Looking for XOpenDisplay in /nix/store/p4i93n4m7al2anavcz54cybw9hy2pik3-libX11-1.8.7/lib/libX11.so;/nix/store/drnbcwzvha7mrb5yxzyrdy9a75c4gjgq-libXext-1.3.4/lib/libXext.so - found
-- Looking for gethostbyname
-- Looking for gethostbyname - found
-- Looking for connect
-- Looking for connect - found
-- Looking for remove
-- Looking for remove - found
-- Looking for shmat
-- Looking for shmat - found
-- Looking for IceConnectionNumber in ICE
-- Looking for IceConnectionNumber in ICE - found
-- Found PkgConfig: /nix/store/7l5gbdz3rcab57r24arg61i4vavpf2cz-pkg-config-wrapper-0.29.2/bin/pkg-config (found version "0.29.2") 
-- XCB[XCB]: Found component XCB
-- Found XCB: /nix/store/2d3v799icb8x8bf63n941b87aalsvsis-libxcb-1.14/lib/libxcb.so  found components: XCB 
-- Found X11_XCB: /nix/store/p4i93n4m7al2anavcz54cybw9hy2pik3-libX11-1.8.7/lib/libX11-xcb.so  
-- Found OpenGL: /nix/store/gr5qpcswiyvg69xriylihlil1amv7fdi-libGL-1.6.0/lib/libOpenGL.so  found components: EGL 
-- Found Wayland: /nix/store/0597w4kkj5x9kryscks4g5pmj35f1vbs-wayland-1.22.0/lib/libwayland-client.so;/nix/store/0597w4kkj5x9kryscks4g5pmj35f1vbs-wayland-1.22.0/lib/libwayland-server.so;/nix/store/0597w4kkj5x9kryscks4g5pmj35f1vbs-wayland-1.22.0/lib/libwayland-egl.so;/nix/store/0597w4kkj5x9kryscks4g5pmj35f1vbs-wayland-1.22.0/lib/libwayland-cursor.so   
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Found FFmpeg: /nix/store/xjzvncwg01pvwq7nz08lcfsgibnkg565-ffmpeg-full-5.1.3-lib/lib/libavformat.so (found version "#define LIBAVFORMAT_VERSION_MINOR  27;#define LIBAVFORMAT_VERSION_MICRO 100.27.100") found components: avformat avutil swscale swresample avcodec 
-- Found ZLIB: /nix/store/8mw6ssjspf8k1ija88cfldmxlbarl1bb-zlib-1.2.13/lib/libz.so (found version "1.2.13") 
-- Performing Test HAVE_STDATOMIC
-- Performing Test HAVE_STDATOMIC - Success
-- OBS:  Qt version: 6
CMake Error at libobs/CMakeLists.txt:408 (find_package):
  By not providing "FindPulseAudio.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "PulseAudio", but CMake did not find one.

  Could not find a package configuration file provided by "PulseAudio" with
  any of the following names:

    PulseAudioConfig.cmake
    pulseaudio-config.cmake

  Add the installation prefix of "PulseAudio" to CMAKE_PREFIX_PATH or set
  "PulseAudio_DIR" to a directory containing one of the above files.  If
  "PulseAudio" provides a separate development package or SDK, be sure it has
  been installed.


-- Configuring incomplete, errors occurred!
See also "/build/source/build/CMakeFiles/CMakeOutput.log".
error: builder for '/nix/store/y2342xfbbvvgynb19ynqqj1fq3ydsa34-obs-studio-29.0.2.drv' failed with exit code 1
error: 1 dependencies of derivation '/nix/store/mr075kkpanl3568y336ydvbc3a7d6zsy-system-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/qi0spjkhk8g7yizjirp3mw0hafq2z39j-nixos-system-Simon-s-Tuxedo-InfinityBook-14-Gen8-23.05.5125.0561103cedb1.drv' failed to build
```

### Notify maintainers

@jb55, @MP2E, @deviant, @materusPL 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.63, NixOS, 23.05 (Stoat), 23.05.5125.0561103cedb1`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.6`
 - channels(root): `"nixos-23.05, nixos-unstable"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

--- a/pkgs/applications/video/obs-studio/default.nix
+++ b/pkgs/applications/video/obs-studio/default.nix
@@ -162,6 +162,9 @@ stdenv.mkDerivation (finalAttrs: {
     "-DENABLE_JACK=ON"
     (lib.cmakeBool "ENABLE_QSV11" stdenv.hostPlatform.isx86_64)
     (lib.cmakeBool "ENABLE_LIBFDK" withFdk)
+    (lib.cmakeBool "ENABLE_ALSA" alsaSupport)
+    (lib.cmakeBool "ENABLE_PULSEAUDIO" pulseaudioSupport)
+    (lib.cmakeBool "ENABLE_PIPEWIRE" pipewireSupport)
   ];
 
   dontWrapGApps = true;

--- a/pkgs/applications/video/obs-studio/default.nix
+++ b/pkgs/applications/video/obs-studio/default.nix
@@ -202,7 +202,7 @@ stdenv.mkDerivation (finalAttrs: {
       video content, efficiently
     '';
     homepage = "https://obsproject.com";
-    maintainers = with maintainers; [ jb55 MP2E materus fpletz ];
+    maintainers = with maintainers; [ eclairevoyant jb55 MP2E materus fpletz ];
     license = with licenses; [ gpl2Plus ] ++ optional withFdk fraunhofer-fdk;
     platforms = [ "x86_64-linux" "i686-linux" "aarch64-linux" ];
     mainProgram = "obs";

